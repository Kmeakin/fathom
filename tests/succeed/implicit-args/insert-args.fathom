let id = fun (@A : Type) (a : A) => a;
let always = fun (@A : Type) (@B : Type) (a : A) (b : B) => a;

let _ = id false;
let _ = always (0 : U32) false;
{}
