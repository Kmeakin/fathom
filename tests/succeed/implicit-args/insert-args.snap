stdout = '''
let id : fun (@A : Type) -> A -> A = fun @A a => a;
let always : fun (@A : Type) (@B : Type) -> A -> B -> A = fun @A @B a b => a;
let _ : Bool = id @Bool false;
let _ : U32 = always @U32 @Bool 0 false;
() : ()
'''
stderr = ''
